2020-09-24 19:55:00,547 ERROR: Exception on /bet [POST] [in /home/guido/anaconda3/envs/local/lib/python3.8/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/home/guido/Dropbox/cs50x/final_project/app/routes.py", line 180, in bet
    bet_matches = Bets.query.filter_by(user_id=User.query.filter_by(username=current_user).first().id).all()
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 1893, in filter_by
    clauses = [
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 1894, in <listcomp>
    _entity_descriptor(self._joinpoint_zero(), key) == value
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 365, in __eq__
    return self.operate(eq, other)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/orm/attributes.py", line 219, in operate
    return op(self.comparator, *other, **kwargs)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 365, in __eq__
    return self.operate(eq, other)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/orm/properties.py", line 367, in operate
    return op(self.__clause_element__(), *other, **kwargs)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/annotation.py", line 104, in __eq__
    return self.__element.__class__.__eq__(self, other)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 365, in __eq__
    return self.operate(eq, other)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 759, in operate
    return op(self.comparator, *other, **kwargs)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 365, in __eq__
    return self.operate(eq, other)
  File "<string>", line 1, in <lambda>
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/type_api.py", line 67, in operate
    return o[0](self.expr, op, *(other + o[1:]), **kwargs)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/default_comparator.py", line 105, in _boolean_compare
    obj = _check_literal(expr, op, obj)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/default_comparator.py", line 373, in _check_literal
    return expr._bind_param(operator, other, type_=bindparam_type)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 4219, in _bind_param
    return BindParameter(
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 1230, in __init__
    self.type = _compared_to_type.coerce_compared_value(
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/type_api.py", line 578, in coerce_compared_value
    _coerced_type = _resolve_value_to_type(value)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/sqltypes.py", line 3024, in _resolve_value_to_type
    raise exc.ArgumentError(
sqlalchemy.exc.ArgumentError: Object <User guido> is not legal as a SQL literal value
2020-09-24 19:56:45,391 INFO: Futbolero startup [in /home/guido/Dropbox/cs50x/final_project/app/__init__.py:37]
2020-09-24 19:56:53,620 INFO: Futbolero startup [in /home/guido/Dropbox/cs50x/final_project/app/__init__.py:37]
2020-09-24 19:58:14,384 INFO: Futbolero startup [in /home/guido/Dropbox/cs50x/final_project/app/__init__.py:37]
2020-09-24 19:59:59,447 ERROR: Exception on /bet [POST] [in /home/guido/anaconda3/envs/local/lib/python3.8/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/home/guido/Dropbox/cs50x/final_project/app/routes.py", line 184, in bet
    new_user_bets = Bets(user_id=User.query.filter_by(username=current_user).first().id,
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 1893, in filter_by
    clauses = [
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 1894, in <listcomp>
    _entity_descriptor(self._joinpoint_zero(), key) == value
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 365, in __eq__
    return self.operate(eq, other)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/orm/attributes.py", line 219, in operate
    return op(self.comparator, *other, **kwargs)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 365, in __eq__
    return self.operate(eq, other)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/orm/properties.py", line 367, in operate
    return op(self.__clause_element__(), *other, **kwargs)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/annotation.py", line 104, in __eq__
    return self.__element.__class__.__eq__(self, other)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 365, in __eq__
    return self.operate(eq, other)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 759, in operate
    return op(self.comparator, *other, **kwargs)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/operators.py", line 365, in __eq__
    return self.operate(eq, other)
  File "<string>", line 1, in <lambda>
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/type_api.py", line 67, in operate
    return o[0](self.expr, op, *(other + o[1:]), **kwargs)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/default_comparator.py", line 105, in _boolean_compare
    obj = _check_literal(expr, op, obj)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/default_comparator.py", line 373, in _check_literal
    return expr._bind_param(operator, other, type_=bindparam_type)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 4219, in _bind_param
    return BindParameter(
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 1230, in __init__
    self.type = _compared_to_type.coerce_compared_value(
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/type_api.py", line 578, in coerce_compared_value
    _coerced_type = _resolve_value_to_type(value)
  File "/home/guido/anaconda3/envs/local/lib/python3.8/site-packages/sqlalchemy/sql/sqltypes.py", line 3024, in _resolve_value_to_type
    raise exc.ArgumentError(
sqlalchemy.exc.ArgumentError: Object <User paula> is not legal as a SQL literal value
2020-09-24 20:00:15,582 INFO: Futbolero startup [in /home/guido/Dropbox/cs50x/final_project/app/__init__.py:37]
